be/src/exec/query_cache/cache_operator.cpp

```diff
@@ -237,8 +246,9 @@ void CacheOperator::_handle_stale_cache_value(int64_t tablet_id, CacheValue& cac
     Version delta_versions(min_version, max_version);
     buffer->tablet = tablet;
     auto has_delete_predicates = tablet->has_delete_predicates(delta_versions);
-    // case 1: there exist delete predicates in delta versions, the cache result is not reuse, so cache miss.
-    if (has_delete_predicates) {
+    // case 1: there exist delete predicates in delta versions, or data model can not support multiversion cache and
+    // the tablet has non-empty delta rowsets; then cache result is not reuse, so cache miss.
+    if (has_delete_predicates || (!_cache_param.can_use_multiversion && !all_rs_empty)) {
```

Comment:
`has_delete_predicates ` is supported when `all_rs_empty `  is true?

Reply:
It is just two cases can be handled in cache-miss mode, to see the comment